ocean_spark/hooks.py

```diff
@@ -92,37 +95,38 @@ def _do_api_call(
                 )
                 response.raise_for_status()
                 return response.json()
+            except (
+                requests_exceptions.ConnectionError,
+                requests_exceptions.Timeout,
+            ) as e:
+                self.log.error(
+                    "Request attempt [%d] to the Ocean Spark API failed with reason: %s",
+                    attempt_num,
+                    str(e),
+                )
             except requests_exceptions.RequestException as e:
-                if not _retryable_error(e):
-                    # In this case, the user probably made a mistake.
-                    # Don't retry.
-                    raise AirflowException(
-                        "Response: {0}, Status Code: {1}".format(
-                            e.response.content, e.response.status_code
-                        )
-                    )
-
-                self._log_request_error(attempt_num, e)
+                msg = self._construct_error_message(e)
+                self.log.error(msg)
+                raise AirflowException(msg)
 
             if attempt_num == self.retry_limit:
                 raise AirflowException(
-                    (
-                        "API requests to Data Mechanics failed {} times. "
-                        + "Giving up."
-                    ).format(self.retry_limit)
+                    f"Request to Ocean Spark API failed failed {self.retry_limit} times. Giving up."
                 )
 
             attempt_num += 1
             time.sleep(self.retry_delay.total_seconds())
 
-    def _log_request_error(
-        self, attempt_num: int, error: requests_exceptions.RequestException
-    ) -> None:
-        self.log.error(
-            "Attempt %s API Request to Data Mechanics failed with reason: %s",
-            attempt_num,
-            error,
-        )
+    @staticmethod
+    def _construct_error_message(ex: requests_exceptions.RequestException) -> str:
+        try:
+            api_response: ApiResponse = json.loads(ex.response.content)
```

Comment:
Are you sure the response is always going to contain valid JSON?

For example, what if the HTTP error code comes from a load balancer on top of the optimizer whose connection to the optimizer timed out?